tests/integration/targets/aci_l3out_logical_node_profile_to_node/tasks/main.yml

```diff
@@ -0,0 +1,160 @@
+  cisco.aci.aci_l3out:
+    <<: *vrf_present
+    l3out: l3outtest
+    domain: l3outtest
+    route_control: export 
+
+- name: Crete node profile
+  cisco.aci.aci_l3out_logical_node_profile: &np_present
+    <<: *tenant_present
+    l3out: l3outtest
+    node_profile: NODES
+
+# BEGIN WITH TESTS
+- name: Add node (check_mode)
+  cisco.aci.aci_l3out_logical_node_profile_to_node: &node_present
+    <<: *np_present
+    pod_id: 1
+    node_id: 111
+    router_id: 111.111.111.111
+    router_id_as_loopback: 'no'
+  check_mode: yes
+  register: cm_add_node
+
+- name: Add node (normal mode)
+  cisco.aci.aci_l3out_logical_node_profile_to_node: 
+    <<: *node_present
+  register: nm_add_node
+
+- name: Verify nm_add_node
+  assert:
+    that:
+    - cm_add_node is changed
+    - nm_add_node is changed
+    - cm_add_node.previous == nm_add_node.previous == []
+    - cm_add_node.sent.l3extRsNodeL3OutAtt.attributes.rtrId == nm_add_node.sent.l3extRsNodeL3OutAtt.attributes.rtrId == '111.111.111.111'
+    - cm_add_node.sent.l3extRsNodeL3OutAtt.attributes.rtrIdLoopBack == nm_add_node.sent.l3extRsNodeL3OutAtt.attributes.rtrIdLoopBack == 'no'
+    - cm_add_node.sent.l3extRsNodeL3OutAtt.attributes.tDn == nm_add_node.sent.l3extRsNodeL3OutAtt.attributes.tDn == 'topology/pod-1/node-111'
+
+- name: Add node  again, check if idempotency works
+  cisco.aci.aci_l3out_logical_node_profile_to_node: 
+    <<: *node_present
+  register: add_node_again
+
+- name: Verify add_node_again
+  assert:
+    that:
+    - add_node_again is not changed
+
+# UPDATE NODE
+- name: Change roouter id
+  cisco.aci.aci_l3out_logical_node_profile_to_node: &node_update
+    <<: *node_present
+    router_id: 11.11.11.11
+  register: update_node
+
+- name: Verify update_node
+  assert:
+    that:
+    - update_node is changed
+    - update_node.previous != []
+    - update_node.sent.l3extRsNodeL3OutAtt.attributes.rtrId == '11.11.11.11'
+
+# ADD ANOTHER NODE
+- name: Add another node
+  cisco.aci.aci_l3out_logical_node_profile_to_node: 
+    <<: *node_present
+    node_id: 112
+    router_id: 12.12.12.12
+
+# QUERY ALL NODES
+- name: Query all nodes
+  cisco.aci.aci_l3out_logical_node_profile_to_node: 
+    <<: *aci_info
+    state: query
+  register: query_all_nodes
+
+- name: Verify query_all_nodes
+  assert:
+    that:
+    - query_all_nodes is not changed
+    - query_all_nodes.current|length >= 1
+
+# QUERY A SPECIFIC NODE
+- name: Query a specific node
+  cisco.aci.aci_l3out_logical_node_profile_to_node:
+    <<: *node_update
+    state: query
+  register: query_spec_node
+
+- name: Verify query_spec_node
+  assert:
+    that:
+    - query_spec_node is not changed
+    - query_spec_node.current|length == 1
+
+# REMOVE NODE
+- name: Remove node
+  cisco.aci.aci_l3out_logical_node_profile_to_node:
+    <<: *node_update
+    state: absent
+  register: remove_node
+
+- name: Verify remove_node
+  assert:
+    that:
+    - remove_node is changed
+    - remove_node.current == []
+
+# CLEANUP 
```

Comment:
Cleanup should preferably be at the start and not at the end as the end might not get executed in case the module breaks in between.

Reply:
Deleted cleanup at the end (there is already  a cleanup at the start)